### PR TITLE
feat: Function to add parameters to a node

### PR DESCRIPTION
**Pull request by kobayu858**
###SUMMARY
-Function to add parameters to a node

### CODE REVIEW

File: lib/src/graph_extension.rs [hunk 1]
```diff
@@ -49,6 +51,25 @@ pub trait GraphExtension {
 }
 
 impl GraphExtension for Graph<NodeData, f32> {
+    fn add_params(&mut self, node: NodeIndex, key: &str, value: f32) {
+        let node_added = self.node_weight_mut(node).unwrap();
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:55]:
```suggestion
        let target_node = self.node_weight_mut(node).unwrap();
```

File: lib/src/graph_extension.rs [hunk 2]
```diff
@@ -49,6 +51,25 @@ pub trait GraphExtension {
 }
 
 impl GraphExtension for Graph<NodeData, f32> {
+    fn add_params(&mut self, node: NodeIndex, key: &str, value: f32) {
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:54]:
```suggestion
    fn add_params(&mut self, node_i: NodeIndex, key: &str, value: f32) {
```

File: lib/src/graph_extension.rs [hunk 3]
```diff
@@ -49,6 +51,25 @@ pub trait GraphExtension {
...
+        let node_added = self.node_weight_mut(node).unwrap();
+        if node_added.params.contains_key(key) {
+            warn!("The key already exists. key: {}", key);
+        } else {
+            node_added.params.insert(key.to_string(), value);
+        }
+    }
+
+    fn update_params(&mut self, node: NodeIndex, key: &str, value: f32) {
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:63]:
```suggestion
    fn update_params(&mut self, node_i: NodeIndex, key: &str, value: f32) {
```

File: lib/src/graph_extension.rs [hunk 4]
```diff
@@ -49,6 +51,25 @@ pub trait GraphExtension {
...
+        let node_added = self.node_weight_mut(node).unwrap();
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:64]:
```suggestion
        let target_node = self.node_weight_mut(node).unwrap();
```

File: lib/src/graph_extension.rs [hunk 5]
```diff
@@ -49,6 +51,25 @@ pub trait GraphExtension {
...
+        println!("node_added: {:?}", !node_added.params.contains_key(key));
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:65]: Remove

File: lib/src/graph_extension.rs (hunk 3 above)
Comment (atsushi421) [inline lib/src/graph_extension.rs:63]:
```suggestion
    fn update_param(&mut self, node: NodeIndex, key: &str, value: f32) {
```

File: lib/src/graph_extension.rs (hunk 2 above)
Comment (atsushi421) [inline lib/src/graph_extension.rs:54]:
```suggestion
    fn add_param(&mut self, node: NodeIndex, key: &str, value: f32) {
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/graph_extension.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:55]: Fixed.

File: lib/src/graph_extension.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:54]: Fixed.

File: lib/src/graph_extension.rs (hunk 3 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:63]: Fixed.

File: lib/src/graph_extension.rs (hunk 4 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:64]: Fixed.

File: lib/src/graph_extension.rs (hunk 5 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:65]: Fixed.

File: lib/src/graph_extension.rs (hunk 3 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:63]: Fixed.

File: lib/src/graph_extension.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:54]: Fixed.

File: lib/src/graph_extension.rs [hunk 6]
```diff
@@ -26,6 +26,8 @@ impl NodeData {
 }
 
 pub trait GraphExtension {
+    fn add_param(&mut self, node: NodeIndex, key: &str, value: i32);
+    fn update_param(&mut self, node: NodeIndex, key: &str, value: i32);
```
Comment (atsushi421) [inline lib/src/graph_extension.rs:30]:
```suggestion
    fn add_param(&mut self, node_i: NodeIndex, key: &str, value: i32);
    fn update_param(&mut self, node_i: NodeIndex, key: &str, value: i32);
```

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/graph_extension.rs (hunk 6 above)
Comment (kobayu858) [reply lib/src/graph_extension.rs:30]: Fixed.

**Review by atsushi421 — APPROVED** [review]
LGTM!

**Pull request merged**